scripts.py

```diff
@@ -1269,77 +1267,114 @@ def make_query(self, _db, identifier_type, identifiers, log=None):
-            #identifier_ids = [x.id for x in param_args.identifiers]
+            identifier_ids = [x.id for x in param_args.identifiers]
 
         self.query = self.make_query(
             self._db, param_args.identifier_type, param_args.identifiers, self.log
         )
 
         editions = True
         offset = 0
+        output = "ContributorID|\tSortName|\tDisplayName|\tComputedSortName|\tMatchRatio_Etc|\tResolution"
+        print output.encode("utf8")
+
         while editions:
-            editions = self.query.offset(offset).limit(batch_size).all()
+            my_query = self.query.offset(offset).limit(batch_size)
+            editions = my_query.all()
+
             for edition in editions:
                 if edition.contributions:
                     for contribution in edition.contributions:
-                        self.process_contribution(self._db, contribution, edition)
+                        self.process_contribution(self._db, contribution, self.log)
             offset += batch_size
+
             self._db.commit()
         self._db.commit()
 
 
     @classmethod
-    def process_contribution(cls, _db, contribution, edition):
-        if not contribution or not edition:
+    def process_contribution(cls, _db, contribution, log=None):
+        if not contribution or not contribution.edition:
             return
 
         contributor = contribution.contributor
 
-        identifier = edition.primary_identifier
+        identifier = contribution.edition.primary_identifier
+
+        if contributor.sort_name and contributor.display_name:
+            
+            computed_sort_name_local_new = unicodedata.normalize("NFKD", unicode(display_name_to_sort_name(contributor.display_name)))
+            # Did HumanName parser produce a differet result from the plain comma replacement?
+            if (contributor.sort_name.strip().lower() != computed_sort_name_local_new.strip().lower()):
+                # computed names don't match.  by how much?  if it's a matter of a comma or a misplaced 
+                # suffix, we can fix without asking for human intervention.  if the names are very different, 
+                # there's a chance the sort and display names are different on purpose, s.a. when foreign names 
+                # are passed as translated into only one of the fields, or when the author has a popular pseudonym. 
+                # best ask a human.
+
+                # if the relative lengths are off than by a stray space or comma, ask a human
+                # it probably means that a human metadata professional had added an explanation/expansion to the 
+                # sort_name, s.a. "Bob A. Jones" --> "Bob A. (Allan) Jones", and we'd rather not replace this data 
+                # with the "Jones, Bob A." that the auto-algorigthm would generate.
+                length_difference = len(contributor.sort_name.strip()) - len(computed_sort_name_local_new.strip())
```

Comment:
Would it make sense to calculate Levenshtein distance here instead of the length of the strings? I don't like what seems to happen here if the names are totally different but the same length.